core/templates/pages/contributor-dashboard-admin-page/contributor-admin-dashboard-page.component.ts

```diff
@@ -240,23 +245,86 @@ export class ContributorAdminDashboardPageComponent implements OnInit {
     this.languageDropdownShown = !this.languageDropdownShown;
   }
 
-  toggleActivityDropdown(): void {
-    this.activityDropdownShown = !this.activityDropdownShown;
+  getDateThatIsDaysBeforeToday(numberOfDaysBeforeToday: number): Date {
+    const today = new Date();
+    return new Date(
+      today.getTime() - 24 * 60 * 60 * 1000 * numberOfDaysBeforeToday
+    );
+  }
+
+  isValidStartDate(selectedStartDate: Date): boolean {
+    const today = new Date();
+    if (
+      today.getTime() >= selectedStartDate.getTime() &&
+      selectedStartDate.getTime() >= this.endDateForUsersLastActivity.getTime()
+    ) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  isValidEndDate(selectedEndDate: Date): boolean {
+    const today = new Date();
+    if (
+      today.getTime() >= selectedEndDate.getTime() &&
+      selectedEndDate.getTime() <= this.startDateForUsersLastActivity.getTime()
+    ) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  selectFirstActivity(): number {
+    const oneDay = 24 * 60 * 60 * 1000;
+    const today = new Date();
+    return Math.floor(
+      Math.abs(
+        today.getTime() - this.startDateForUsersLastActivity?.getTime()
+      ) / oneDay
+    );
+  }
+
+  selectLastActivity(): number {
```

Comment:
`getNumberOfDaysSinceStartDateForLastActivity` and `getNumberOfDaysSinceEndDateForLastActivity`